ion-hash/src/lib.rs

```diff
@@ -141,3 +127,22 @@ fn ion_hash_escape(representation: &[u8]) -> Vec<u8> {
 
     out
 }
+
+// TODO: Use the one below after some refactoring (it currently doesn't support
+// just writing the inner representation).
+fn binary_repr<E: Element + ?Sized>(elem: &E) -> IonResult<Vec<u8>> {
+    Ok(match elem.ion_type() {
+        IonType::String => Vec::from(elem.as_str().unwrap().as_bytes()),
+        _ => todo!(),
+    })
+}
+
+fn _binary_repr<E: Element + ?Sized>(elem: &E) -> IonResult<Vec<u8>> {
+    // TODO: Fix this once the writer API supports non-fixed-size buffers.
+    let mut buf = vec![0u8; 4096];
+    let mut writer = Format::Binary.element_writer_for_slice(&mut buf)?;
+    writer.write(elem)?;
+    let len = writer.finish()?.len();
+    buf.truncate(len);
+    Ok(buf)
+}
```

Comment:
Do you want the full Ion binary encoding including the IVM + symbol table?  I don't think that is right.

Reply:
Nope. The other implementations factor out the binary writer and the "raw" writer. 

https://github.com/amzn/ion-hash-java/blob/b855f0836a527b05cca2895f1099c0028283c0a9/src/com/amazon/ionhash/HasherImpl.java#L353
https://github.com/amzn/ion-java/blob/0eb62ba92633f03e9f3175bb77f1b3214fcc1cdd/src/com/amazon/ion/impl/bin/_PrivateIon_HashTrampoline.java#L35
https://github.com/amzn/ion-java/blob/0eb62ba92633f03e9f3175bb77f1b3214fcc1cdd/src/com/amazon/ion/impl/bin/IonRawBinaryWriter.java#L1085

Reply:
Yeah, I would feel more comfortable just ripping this dead code out for now...  The crux is you want the raw encoding functions, and we should figure that problem out.

Reply:
yeah! For now, I'm just going to duplicate them by hand since it's not very complicated.

Latest commit adds ints and removes this code.